notebooks/Notebook1.md

```diff
@@ -2,33 +2,94 @@
+ - [Executing Shell Commands in Python](https://stackabuse.com/executing-shell-commands-with-python/)
+ - [Convert Python To Executable](https://towardsdatascience.com/how-to-easily-convert-a-python-script-to-an-executable-file-exe-4966e253c7e9)
+ - [Scalapy -- Running Python packages in Scala](https://github.com/scalapy/scalapy)
+ - [Turning and Exporting a Scala Project as One single .jar file](https://github.com/sbt/sbt-onejar)
+ - [GUI programming in Scala](https://otfried.org/scala/gui.html)
+
+
+### Thoughts: 
+
+From my research the actual structure of my program, I chose to do the following: The DSL will be a External Domain language, and implemented in scala. While I did a internal one for the previous project, I will have some guidence from others that did external while I set that up. Then, the external DSL will be set into a simple GUI following the tutorial above. Im not going to focus on prettyness or anything like that, I imagine this will be simple and I will focus mostly on the implementation of the language. The gui will either allow directly editing a text document and a run button, or to select a already written text file. 
+
+The language will have two overarcing setups. 1 build a roll roll {description} and then enacting stats on that roll. Rolls can be reused by a name and can be nested if possible. See design for more. The creation of stats would use the Scalapy package to grab some python graphing packages. There might be some scala graphing packages tho, so Ill keep an eye out for that. 
+
+Lastly, I will convert the scala to a single jar file as above. So the final checklist is 
+- Make the simplest working version of the DSL 
+- Make a Simple *working* GUI
+- Set up Jar adaptation 
+- Add Stats to DSL
+- Improve DSL 
+- Improve GUI 
+
+### Design :
+
+This is a preliminary design but this is what I want people to be able to create and explore with the system. 
+Making a roll: (basically a function that outputs a number) 
+```
+roll alpha {
+
+1d6 + 7
+
+}
+```
+note: 1d6 evaluates to a roll object. 
+
+Making a check: (basically a fucntion that outputs a succes or failure -- maybe add allowance for multiple successes reported?) 
+```
+check beta {
+
+5d6 if >10 
+
+}
+```
+Making a stat: (output stats with various parameters) 
+```
+roll gamma {
+ alpha + 1d6
+}
+
+stat beta (parameters)
+stat gamma (parameters)
+```
+This still needs some refinement. I want to break away from coding conventions, but we know thats very difficult. 
+
+
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+I feel much better about starting now. The most pressing issue is I have is I dont have experience with external implementations. Something my DSL relies on. Hopefully I can get guidence from the rest of the class on how that works. 
```

Comment:
Yes, and let me and Gabriel know (e.g., on Discord) if you would like some guidance. Are you looking for tools that you can use to implement a grammar, or...?